app/models/invite.rb

```diff
@@ -176,30 +242,64 @@ def self.get_group_ids(group_names)
     group_ids
   end
 
-  def self.find_all_invites_from(inviter, offset = 0, limit = SiteSetting.invites_per_page)
-    Invite.where(invited_by_id: inviter.id)
+  def self.find_all_pending_invites_from(inviter, offset = 0, limit = SiteSetting.invites_per_page)
+    Invite.single_use_invites
+      .includes(:invited_users)
+      .includes(invited_users: :user)
+      .where(invited_by_id: inviter.id)
       .where('invites.email IS NOT NULL')
-      .includes(user: :user_stat)
-      .order("CASE WHEN invites.user_id IS NOT NULL THEN 0 ELSE 1 END, user_stats.time_read DESC, invites.redeemed_at DESC")
+      .where('invited_users.user_id IS NULL')
+      .order('invites.updated_at DESC')
       .limit(limit)
       .offset(offset)
-      .references('user_stats')
+      .references('invited_users')
+      .references('user')
   end
 
   def self.find_pending_invites_from(inviter, offset = 0)
-    find_all_invites_from(inviter, offset).where('invites.user_id IS NULL').order('invites.updated_at DESC')
+    find_all_pending_invites_from(inviter, offset)
   end
 
-  def self.find_redeemed_invites_from(inviter, offset = 0)
-    find_all_invites_from(inviter, offset).where('invites.user_id IS NOT NULL').order('invites.redeemed_at DESC')
+  def self.find_pending_invites_count(inviter)
+    find_all_pending_invites_from(inviter, 0, nil).reorder(nil).count
   end
 
-  def self.find_pending_invites_count(inviter)
-    find_all_invites_from(inviter, 0, nil).where('invites.user_id IS NULL').reorder(nil).count
+  def self.find_all_redeemed_invites_from(inviter, offset = 0, limit = SiteSetting.invites_per_page)
+    InvitedUser.includes(:invite)
+      .includes(user: :user_stat)
+      .includes(:user)
+      .where('invited_users.user_id IS NOT NULL')
```

Comment:
I checked and `invited_users.user_id IS NOT NULL` is only used here and nowhere else. Can you point out where else did you see this condition?

Also `invited_users.user_id IS NOT NULL` is not same as `.single_use_invites` scope because it tells which invites have been redeemed. This condition is required to show all the redeemed invites from a particular user.

Reply:
Oh yes you're right. I confused it with `invited_users.user_id IS NULL` :+1: